Datastore/src/Operation.php

```diff
@@ -280,11 +316,30 @@ public function beginTransaction($transactionOptions, array $options = [])
                 $transactionOptions['readOnly']
             );
         }
-        $res = $this->connection->beginTransaction($options + [
+
+        array_walk($transactionOptions, function (&$item) {
```

Comment:
A comment will be helpful here.
And why was this needed for v2?

Reply:
This logic was there in gRPC to comply with the modification present in https://github.com/googleapis/google-cloud-php/blob/main/Datastore/src/DatastoreClient.php#L560-L563